wasatch/SPIDevice.py

```diff
@@ -0,0 +1,331 @@
+import re
+import os
+import usb
+import time
+import json
+import struct
+import logging
+import datetime
+from queue import Queue
+from typing import Callable, Any
+
+# Needed for Mac side
+# Device Finder should already have done this
+# For thoroughness though doing here anyway
+# This is required for finding the usb <-> serial board
+import usb.core
+usb.core.find()
+
+from .SpectrometerSettings        import SpectrometerSettings
+from .SpectrometerState           import SpectrometerState
+from .SpectrometerResponse        import SpectrometerResponse
+from .SpectrometerRequest         import SpectrometerRequest
+from .SpectrometerResponse        import ErrorLevel
+from .ControlObject               import ControlObject
+from .DeviceID                    import DeviceID
+from .InterfaceDevice             import InterfaceDevice
+from .Reading                     import Reading
+from .EEPROM                      import EEPROM
+
+log = logging.getLogger(__name__)
+
+class SPIDevice(InterfaceDevice):
```

Comment:
Let's add a @todo that we still need to add:

- CRC validation
- set_gain_db
- set_pixel_mode (i.e. detector resolution)
- set_start_line
- set_stop_line
- set_start_column
- set_stop_column
- set_trigger_mode (not currently documented in ENG-0150)

Pretty much all of those will be needed for near-term testing.